### ISSUE TITLE
[Bug] Missing find match for $$

### ISSUE THREAD

**Issue by antshar** [texstudio-org/texstudio#2475]
In previous versions of TeXStudio, `$...$` and `$$...$$` were treated as matching group like it does with environments names. This way I could use all the insanely useful shortcuts:

![image](https://user-images.githubusercontent.com/54121465/178960681-10767e74-5e4b-47c2-9e62-1bd07228980d.png)

Here's a footage made in 2.12.22 version

https://user-images.githubusercontent.com/54121465/178960904-d2f847fe-9c3e-43a7-83fe-e969a3274860.mp4

And here is footage made in 4.3.0beta2

https://user-images.githubusercontent.com/54121465/178961014-2f95d8fc-70cb-41b7-8586-76dc7561204c.mp4

Could you please return that behavior for `$` and `$$`?

**Comment by sunderme** [comment]
duplicate to #1312

**Comment by antshar** [comment]
Finally, after combing through commits, I found what causes it: https://github.com/texstudio-org/texstudio/commit/adb7012e0513e9e50c150b05719b7ea0a4623c67

Some syntax highlighting have been moved to source code that is being hardcoded now, however you can revert the treatment of `$` and `$$` as parentheses **even yourself** without recompiling the program!

It's the matter of adding these lines
```
<sequence id="keywords/single" format="align-ampersand">&amp;</sequence>

<context id="mathmode$$" format="numbers" transparency="true">
  <start parenthesis="math$$:open" parenthesisWeight="50" fold="1" format="math-delimiter" >\$\$</start>
  <stop parenthesis="math$$:close" parenthesisWeight="50" fold="1" format="math-delimiter" >\$\$</stop>
  <stop parenthesis="math$$:close" parenthesisWeight="0" fold="1" format="math-delimiter" >^\n</stop>
</context>

<context id="mathmode$" format="numbers" transparency="true">
  <start parenthesis="math:open" parenthesisWeight="40" fold="1" format="math-delimiter" >\$</start>
  <stop parenthesis="math:close" parenthesisWeight="40" fold="1" format="math-delimiter" >\$</stop>
  <stop parenthesis="math:close" parenthesisWeight="40" fold="1" format="magicComment" >%\$</stop><!-- can be used as a workaround for closing math at the end of the line, if highlighting misinterprets a $ sign, e.g. \path{$PATH} %$ -->
  <stop parenthesis="math:close" parenthesisWeight="0" fold="1" format="math-delimiter" >^\n</stop>
</context>
```
into `tex.qnfa` file that can be found here: https://github.com/texstudio-org/texstudio/blob/master/utilities/qxs/tex.qnfa

Put a modified file with those lines into `config/languages` folder. Note that you might need to create `languages` folder yourself, since by default it is not created. And the `config` folder is the one that portable version has whereas the regular path for it is
```
C:\Users\<NAME>\AppData\Roaming\texstudio
```

I can provide the modified file so you need only to put it into the specified folder and good old `$`, `$$` are now treated as parentheses again and everything works like a charm.

▶ [**tex.qnfa.zip**](https://github.com/texstudio-org/texstudio/files/9127511/tex.qnfa.zip) (unzip it first)

**Comment by antshar** [comment]
@sunderme it's still marked as "help wanted". I found a solution, can you add these changes to the source code?

**Comment by sunderme** [comment]
your solution is not a solution. The lines were intentionally removed from tex.qnfa because a different (faster) syntax highlighting scheme is used. The parenthesis matching is a side effect of this. A fix will keep the current functionality and restore the matching.

**Comment by sunderme** [comment]
fixed

**Comment by antshar** [comment]
@sunderme doesn't seem to be fixed in 4.7.1
